ipfs-controller/controller.go

```diff
@@ -0,0 +1,231 @@
+package ipfsController
+
+import (
+	"context"
+	"errors"
+	"net"
+	"strings"
+	"time"
+
+	"github.com/ipfs-cluster/ipfs-cluster/api"
+	"github.com/ipfs-cluster/ipfs-cluster/api/rest/client"
+	shell "github.com/ipfs/go-ipfs-api"
+	ma "github.com/multiformats/go-multiaddr"
+	"github.com/threefoldtech/tf-pinning-service/config"
+	"github.com/threefoldtech/tf-pinning-service/pinning-api/models"
+)
+
+const DefaultPinMode = api.PinModeRecursive
+
+type clusterController struct {
+	Client client.Client
+	// Attributes
+	ReplicationFactorMin int
+	ReplicationFactorMax int
+	PinMode              api.PinMode
+}
+
+func GetClusterController() (ipfsController, error) {
```

Comment:
Returning a private type from a public method. either the method is also private or if it need to be public it must return a public type